### ISSUE TITLE
Dark and Darker themes ignore TextColor and BackgroundColor

### ISSUE THREAD

**Issue by NooneAtAll3** [sumatrapdfreader/sumatrapdf#4068]
**SumatraPDF version**
- Version 3.5.1 (or at least the one that reports "you use the latest version" when Check_for_Updates is used)

**Describe the bug**
When using non-Light theme, changing TextColor or BackgroundColor doesn't change the corresponding parts of the document

**To Reproduce**
1. Choose Dark(er) theme
2. Open Advanced Options
3. Change BackgroundColor value and save
4. Observe no change happening

**Expected behavior**
Honestly, I don't know what to expect. I can understand how UI-only control can be expected to change pdf appearance and that "detect non-default values" might give complexity in the future  

IMO, ideally, you don't make "Theme" a state. Make it a preset - selecting a theme should *overwrite* BackgroundColor/TextColor value - allowing further customization from there.  
Or maybe add an "overwrite with theme = true" option, so that user won't delete his values accidentally.

Some extreme solution would be to put Themes as customizable objects *in* the settings - that way both 1) custom themes would be supported, while 2) default ones would stay predictable

**Additional context**
I've been struggling with Sumatra, trying to make everything extra dark for myself ("Black theme" sort of)
But right now non-Light themes give green/blue hue to the pdf, while Light theme becomes "Gondor calling for aid" with its white strip on top - situation very unfortunate

Since you already have a mechanism to change UI color based on user input - exposing it in the settings.txt would be a great next step

Thank you for your work

**Comment by GitHubRulesOK** [comment]
In an app like this there can not be content theming simply objects rendered in a scale of colours, hence no way to mess with PDF browser  colours. What is different about SumatraPDF is it can reverse the range for viewing negatives. However the collateral damage is paper is black and ink is white (NOT theming simply inversion).

Repeatedly this has been seen as a means to fake a theme by altering the end points to light grey or dark grey or any other RGB mix but it is not real body Theming which would require every single pixel or line be edited into another colour.

The User Interface is a different aspect as made of 2 parts. the outer part (Title and ScrollBars) controlled by Windows System and the inner part (tabs and sidebar content) controlled by Visual Studio code.

When Pseudo "Themes" are added  the normal body inversion range needs to be altered hence the grey on grey.
One colour fits all, (pleases some of the people, some of the time but not everybody!) in one theme
so a darker range in another theme.

**Comment by fefrei** [comment]
I can confirm this issue. It's a bit frsustrating if you want dark backgrounds everywhere: You can get an inverted PDF view with SumatraPDF in a light theme, or you can turn SumatraPDF dark but your document is no longer inverted :-(

In a perfect world, as @NooneAtAll3 suggested, these settings should be per-theme, so we can have a toggle between normal-colored PDF & light SumatraPDF and dark, inverted PDF & dark SumatraPDF direktly in the UI ❤️

**Comment by pulbitz** [comment]
Until pre-release 3.6.15945, custom TextColor/BackgroundColor worked fine in the Dark/Darker theme.
Since the bug in the Light theme below was fixed, I've been having issues.
https://github.com/sumatrapdfreader/sumatrapdf/issues/4030
It was indeed a bug, but it was a useful bug for Dark/Darker theme users.

**Comment by charygao** [comment]
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/6499936/f3f30779-3913-4dd2-baf6-5bd7e3bf6573)
The **bookmark tree view font** is **hard to identified** in the  darker( or dark ) theme, cause the font color is all most same with the background .   please add the settings 'TreeFontTextColor' \ 'TreeFontTextBackgroundColor' \ 'TreeFontTextSelectionColor' for user . thank you!

**Comment by Cirkutry** [comment]
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/149087292/7d2c5649-4d32-45cc-a165-f467bd25ad59)
Also sub-menu options that are not available for any particular instance are very hard to see in Dark and Darker mode.

**Comment by GitHubRulesOK** [comment]
@Cirkutry that is not a problem they are "greyed out" when inappropriate so with file open and with a favorite available
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f595830f-96e5-4449-9ab5-cebc7ac1a245) ![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/7f296825-3790-44cc-ba8d-421f5e509d53)

**Comment by Cirkutry** [comment]
@GitHubRulesOK Yes, I understand. What I propose is that when a option is seen but not available to use as you illustrated with the "favorite" feature, the color should have a little more contrast. In most applications unavailable options are greyed out but not to the extent of not being visible.

**Comment by GitHubRulesOK** [comment]
Historically main colour was Yellow and grey not a problem with light theme

![image](https://github.com/user-attachments/assets/35f5230a-79d4-4392-9114-2a7e1290683b)

But a lot of people did not like high contrast so it was muted to grey and then everybody insisted it be darker
![image](https://github.com/user-attachments/assets/54f230df-92c5-4cbc-84a5-28d12502f30c)

**Comment by GitHubRulesOK** [comment]
Closing as 
The OP issue at the time was 
> When using non-Light theme, changing TextColor or BackgroundColor doesn't change the corresponding parts of the document

That has been for some time no longer the case

<img width="1482" height="546" alt="Image" src="https://github.com/user-attachments/assets/9b2ea678-3665-424a-8b03-103cd862b9c5" />
